src/examples/environment_renderer.cpp

```diff
@@ -0,0 +1,783 @@
+    glTexParameteri(GL_TEXTURE_CUBE_MAP, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
+    glTexParameteri(GL_TEXTURE_CUBE_MAP, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
+    glTexParameteri(GL_TEXTURE_CUBE_MAP, GL_TEXTURE_WRAP_R, GL_CLAMP_TO_EDGE);
+    
+    // Load each face
+    bool success = true;
+    for (int i = 0; i < 6; i++)
+    {
+      string extension = faceFiles[i].substr(faceFiles[i].find_last_of(".") + 1);
+      transform(extension.begin(), extension.end(), extension.begin(), ::tolower);
+      
+      if (extension == "hdr")
+      {
+        // For now, HDR loading is not implemented, so we'll create a placeholder
+        cout << "HDR face loading not yet implemented for " << faceFiles[i] << ", using placeholder" << endl;
+        // Create a simple colored face as placeholder
+        vector<unsigned char> data(256 * 256 * 3);
+        // Use different colors for different faces for testing
+        unsigned char r = (i == 0 || i == 1) ? 255 : 64;  // Red for X faces
+        unsigned char g = (i == 2 || i == 3) ? 255 : 64;  // Green for Y faces  
+        unsigned char b = (i == 4 || i == 5) ? 255 : 64;  // Blue for Z faces
+        
+        for (int j = 0; j < 256 * 256; j++)
+        {
+          data[j * 3] = r;
+          data[j * 3 + 1] = g;
+          data[j * 3 + 2] = b;
+        }
+        
+        glTexImage2D(faces[i].target, 0, GL_RGB, 256, 256, 0, GL_RGB, GL_UNSIGNED_BYTE, data.data());
+      }
+      else if (extension == "png" || extension == "jpg" || extension == "jpeg")
+      {
+#ifdef __APPLE__
+        // Use Core Graphics to load PNG/JPG on macOS
+        vector<unsigned char> imageData;
+        int imgWidth, imgHeight, imgChannels;
+        
+        if (loadImageWithCoreGraphics(faceFiles[i], imageData, imgWidth, imgHeight, imgChannels))
+        {
+          // Convert RGBA to RGB if needed
+          if (imgChannels == 4)
+          {
+            vector<unsigned char> rgbData(imgWidth * imgHeight * 3);
+            for (int j = 0; j < imgWidth * imgHeight; j++)
+            {
+              rgbData[j * 3] = imageData[j * 4];         // R
+              rgbData[j * 3 + 1] = imageData[j * 4 + 1]; // G
+              rgbData[j * 3 + 2] = imageData[j * 4 + 2]; // B
+              // Skip alpha channel
+            }
+            glTexImage2D(faces[i].target, 0, GL_RGB, imgWidth, imgHeight, 0, GL_RGB, GL_UNSIGNED_BYTE, rgbData.data());
+          }
+          else
+          {
+            glTexImage2D(faces[i].target, 0, GL_RGB, imgWidth, imgHeight, 0, GL_RGB, GL_UNSIGNED_BYTE, imageData.data());
+          }
+          cout << "Loaded " << extension << " face: " << faceFiles[i] << " (" << imgWidth << "x" << imgHeight << ")" << endl;
+        }
+        else
+        {
+          cout << "Failed to load " << extension << " face: " << faceFiles[i] << ", using placeholder" << endl;
+          // Create a simple colored face as placeholder
```

Comment:
Create a function or method to create faced placeholder, because you wrote these lines multiple times.